packages/dui3/components/model/Receiver.vue

```diff
@@ -0,0 +1,255 @@
+  await store.patchModel(props.modelCard.modelCardId, {
+    selectedVersionId: props.modelCard.latestVersionId
+  })
+  if (props.modelCard.progress)
+    await store.receiveModelCancel(props.modelCard.modelCardId)
+  await store.receiveModel(props.modelCard.modelCardId)
+}
+
+const expiredNotification = computed(() => {
+  if (!props.modelCard.latestVersionId || props.modelCard.hasDismissedUpdateWarning)
+    return
+  if (props.modelCard.latestVersionId === props.modelCard.selectedVersionId) return
+  const notification = {} as ModelCardNotification
+  notification.dismissible = true
+  notification.level = 'warning'
+  notification.text = 'Newer version available!'
+  notification.cta = {
+    name: 'Update',
+    action: receiveLatestVersion
+  }
+  return notification
+})
+
+const receiveResultNotification = computed(() => {
+  if (
+    !props.modelCard.bakedObjectIds ||
+    props.modelCard.displayReceiveComplete !== true
+  )
+    return
+
+  const notification = {} as ModelCardNotification
+  notification.dismissible = true
+  notification.level = 'success'
+  notification.text = 'Model loaded!'
+  notification.report = props.modelCard.report
+  notification.cta = {
+    name: 'Highlight',
+    action: () => {
+      app.$baseBinding.highlightModel(props.modelCard.modelCardId)
+    }
+  }
+  return notification
+})
+
+const errorNotification = computed(() => {
+  if (!props.modelCard.error) return
+  const notification = {} as ModelCardNotification
+  notification.dismissible = true
+  notification.level = 'danger'
+  notification.text = props.modelCard.error
+  notification.report = props.modelCard.report
+  return notification
+})
+
+const { result: versionDetailsResult, refetch } = useQuery(
+  versionDetailsQuery,
+  () => ({
+    projectId: props.modelCard.projectId,
+    modelId: props.modelCard.modelId,
+    versionId: props.modelCard.selectedVersionId
+  }),
+  () => ({
+    clientId: props.modelCard.accountId
+  })
+)
+
+const createdAgoUpdater = useInterval(10_000) // refresh the created ago, and latestversion etc. every 10s
+
+const createdAgo = computed(() => {
+  createdAgoUpdater.value
+  return dayjs(versionDetailsResult.value?.project.model.version.createdAt).from(
+    dayjs()
+  )
+})
+
+const latestVersionCreatedAt = computed(() => {
+  createdAgoUpdater.value
+  return dayjs(props.modelCard.latestVersionCreatedAt).from(dayjs())
+})
+
+onMounted(() => {
+  refetch()
+})
+
+// On initialisation, we check whether there was a never version created while we were offline. If so, flagging this dude as expired.
+watchOnce(versionDetailsResult, async (newVal) => {
+  if (!newVal) return
+  let patchObject = {}
+
+  if (
+    newVal?.project.model.versions.items &&
+    newVal?.project.model.versions.items.length !== 0 &&
+    newVal?.project.model.versions.items[0].id !== props.modelCard.selectedVersionId
+  ) {
+    patchObject = {
+      latestVersionId: newVal?.project.model.versions.items[0].id,
```

Comment:
You've already checked that newVal is defined, the optional chaining operator is redundant